### PR TITLE
Proposal:  Use dedent for nicer multi-line literals

### PR DESCRIPTION
**Pull request by thockin**
What do you think about using this - expecially for places like kubectl where there are a lot of line-wrapped literals.

---

This change is [Reviewable](http://reviewable.k8s.io/reviews/kubernetes/kubernetes/17609)

### CODE REVIEW

**Comment by wojtek-t** [comment]
Nice - I like it!

**Comment by smarterclayton** [comment]
As long as can still have indent inside it if we need it, no issue from me.

**Comment by thockin** [comment]
It just strips the longest whitespace prefix common to all lines.  Indents
should work.

**Comment by derekwaynecarr** [comment]
I like it too.

**Comment by bgrant0607** [comment]
cc @kubernetes/kubectl

**Comment by bgrant0607** [comment]
@smarterclayton Please reassign all PRs you don't intend to review.

**Comment by AdoHe** [comment]
I like it too :)

**Comment by janetkuo** [comment]
Looks good. Just need a rebase.

**Comment by janetkuo** [comment]
We should document this at https://github.com/kubernetes/kubernetes/blob/master/docs/devel/kubectl-conventions.md#command-implementation-conventions

File: pkg/kubectl/cmd/describe.go [hunk 1]
```diff
@@ -38,39 +39,42 @@ type DescribeOptions struct {
...
+		will first check for an exact match on TYPE and NAME_PREFIX. If no such resource
+		exists, it will output details for every resource that has a name prefixed with NAME_PREFIX
+
+		Possible resource types include (case insensitive): pods (po), services (svc),
+		replicationcontrollers (rc), nodes (no), events (ev), limitranges (limits),
+		persistentvolumes (pv), persistentvolumeclaims (pvc), resourcequotas (quota),
+		namespaces (ns), serviceaccounts or secrets.`)
+	describe_example = dedent.Dedent(`
+		# Describe a node
+		$ kubectl describe nodes kubernetes-minion-emt8.c.myproject.internal
```
Comment (kargakis) [inline pkg/kubectl/cmd/describe.go]: Remove the dollar sign

**Comment by thockin** [comment]
@mhrgoog as a first bug

**Comment by MHBauer** [comment]
#26761 is merged.

**Pull request closed without merging**